velox/benchmarks/basic/LikeBenchmark.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "velox/benchmarks/ExpressionBenchmarkBuilder.h"
+#include "velox/functions/lib/Re2Functions.h"
+
+using namespace facebook;
+using namespace facebook::velox;
+using namespace facebook::velox::functions;
+using namespace facebook::velox::functions::test;
+using namespace facebook::velox::memory;
+using namespace facebook::velox;
+
+int main(int argc, char** argv) {
+  folly::Init init(&argc, &argv);
+
+  exec::registerStatefulVectorFunction("like", likeSignatures(), makeLike);
+  ExpressionBenchmarkBuilder benchmarkBuilder;
+  const vector_size_t vectorSize = 1000;
+  auto vectorMaker = benchmarkBuilder.vectorMaker();
+
+  auto substringInput =
```

Comment:
There is a good amount of repetition here. Consider extracting helper function:

```
auto substringInput = makeInput(vectorMaker, vectorSize, [] (auto row) { return ...a_b_c...;});

auto prefixInput = makeInput(vectorMaker, vectorSize, [] (auto row) { return a_b_c...;});

auto suffixInput = makeInput(vectorMaker, vectorSize, [] (auto row) { return ...a_b_c;});
```

or use a pair of booleans to indicate whether to pad strings at the beginning and at end.

```
auto substringInput = makeInput(vectorMaker, vectorSize, true, true);

auto prefixInput = makeInput(vectorMaker, vectorSize, false, true);

auto suffixInput = makeInput(vectorMaker, vectorSize, true, false);
```